app/services/CartServiceImpl.java

```diff
@@ -88,23 +94,35 @@ private void clearFrequency(final String cartId) {
     public void setProductToCart(final Cart cart, final ProductProjection product, final ProductVariant variant, final int frequency) {
         requireNonNull(cart, "'cart' must not be null");
         requireNonNull(product, "'product' must not be null");
-        final Cart clearedCart = clearCart(cart);
-        final AddLineItem action = AddLineItem.of(product.getId(), variant.getId(), frequency);
-        final Cart updatedCart = sphereClient().execute(CartUpdateCommand.of(clearedCart, action)).toCompletableFuture().join();
-
-        final CustomObjectDraft<Integer> draft = CustomObjectDraft.ofUnversionedUpsert(ShopKeys.FREQUENCY, updatedCart.getId(), frequency,
-                new TypeReference<CustomObject<Integer>>() {
-                });
 
-        final CustomObject<Integer> customObject = sphereClient().execute(CustomObjectUpsertCommand.of(draft)).toCompletableFuture().join();
-        Logger.debug("Setting new or update CustomObject: {}", customObject);
+        Optional.of(clearCart(cart)).map((clearedCart) -> {
```

Comment:
`map` is for transformation of values, for example in streams it is a lazy operation - so dangerous of you want to do side-effects.

For Optionals this would be more ideomatic if you want to do sth. if the thing is not null:

```java
Optional.ofNullable(input).ifPresent(clearedCart -> {
  //do the side effects
});